### PR TITLE
[release-1.5] 🌱 Grouped CI related cherry-picks

### PR DESCRIPTION
**Pull request by chrischdi**
This is a manual cherry-pick of:

- #2180
- #2167
- #2172
  - only CI parts
- #2174
- #2211
- #2147
  - only CI parts partially
- removes `contrib/haproxy/openapi/docs` which is full of broken links.

### CODE REVIEW

**Comment by sbueringer** [comment]
Same as on 1.7 & 1.6

**Comment by chrischdi** [comment]
Should all be green now.

/assign @sbueringer @kill

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ Note: These OVAs are not updated for security fixes and it is recommended to alw
...
+|  v1.21.11   |   [ova](https://storage.googleapis.com/capv-images/release/v1.21.11/centos-7-kube-v1.21.11.ova), [sha256](https://storage.googleapis.com/capv-images/release/v1.21.11/centos-7-kube-v1.21.11.ova.sha256)   |   [ova](https://storage.googleapis.com/capv-images/release/v1.21.11/ubuntu-1804-kube-v1.21.11.ova), [sha256](https://storage.googleapis.com/capv-images/release/v1.21.11/ubuntu-1804-kube-v1.21.11.ova.sha256)   |   [ova](https://storage.googleapis.com/capv-images/release/v1.21.11/ubuntu-2004-kube-v1.21.11.ova), [sha256](https://storage.googleapis.com/capv-images/release/v1.21.11/ubuntu-2004-kube-v1.21.11.ova.sha256)| [ova](https://storage.googleapis.com/capv-images/release/v1.21.11/photon-3-kube-v1.21.11.ova), [sha256](https://storage.googleapis.com/capv-images/release/v1.21.11/ubuntu-1804-kube-v1.21.11.ova.sha256) |
```
Comment (sbueringer) [inline README.md:57]: Are we now linking the ubuntu sha for the photon image?

Comment (chrischdi) [reply README.md:57]: uh oh totally missed it the wrong way around :D :thx

**Comment by sbueringer** [comment]
Turns out the linkchecker is also useful in CAPV :)

/lgtm
/approve

**Pull request merged**